bindings/python/pymongoarrow/builders.pyi

```diff
@@ -19,6 +19,39 @@ cdef class _ArrayBuilderBase:
             self.append(value)
 
 
+cdef class StringBuilder(_ArrayBuilderBase):
+    type_marker = _BsonArrowTypes.string
+    cdef:
+        shared_ptr[CStringBuilder] builder
+
+    def __cinit__(self, MemoryPool memory_pool=None):
+        cdef CMemoryPool* pool = maybe_unbox_memory_pool(memory_pool)
+        self.builder.reset(new CStringBuilder(pool))
+
+    cpdef append_null(self):
+        self.builder.get().AppendNull()
+
+    def __len__(self):
+        return self.builder.get().length()
+
+    cpdef append(self, value):
+        if value is None or value is np.nan:
+            self.builder.get().AppendNull()
+        elif isinstance(value, (bytes, str)):
```

Comment:
Should we support coercing bytes to string? `bytes` is a separate type in BSON. Also are there any tests for this behavior?

Reply:
When we call `bson_iter_utf8` it returns a `char *`, which gets converted by Cython to `bytes`.  We are guarded by using `value_t == BSON_TYPE_UTF8` in `bson.pyi`.   We'll use the `BinaryBuilder` for `BSON_TYPE_BINARY`.

Reply:
Hmm okay, so if we only ever call this method with bytes why should we check for None, np.nan, str, and other types here? Can this method be simplified to:
```python
cpdef append(self, value):
    self.builder.get().Append(tobytes(value))
```

Reply:
True, this logic was copied from other builders (which in turn was copied from `pyarrow.lib`).  I'll clean up the others as well to match how we're using them.

Reply:
I wonder if we should do this as a separate ticket. It would be interesting to know if there's a perf improvement for simplifying the builders.

Reply:
Ah, I just saw this, do you want me to revert the last commit?  The builders aren't a public API, and the validation is done from `bson.pyi`, so I think it is a logical change.

Reply:
It's fine, I just didn't want to hold up this review since everything else already LGTM. I'l re review the new changes.